### PR TITLE
Docs: Add example for accessing flow/global data in a ui-template

### PR DESCRIPTION
**Pull request by joepavitt**
## Description

- Adds an explicit mention of the best practice to access `flow`/`global` data inside a `ui-template`
- Also adds some eaders to make finding the `this.$socket.on` and `this.send` functions easier to find

## Related Issue(s)

Adds documentation to cover #1073

### CODE REVIEW

File: docs/nodes/widgets/ui-template.md [hunk 1]
```diff
@@ -85,11 +85,23 @@ When accessing the `msg` variable inside a `<script />` tag, you need to prefix
...
 ### Built-in Functions
 
 We also offer some helper functions for the Node-RED integration too:
 
+#### Sending Data
+
 - `this.send` - Send a message to the Node-RED flow. If a non-Object value is sent, then Dashboard will automatically wrap that into a `msg.payload` object.
+
+#### Receiving Data
+
```
Comment (Steve-Mcl) [inline docs/nodes/widgets/ui-template.md:104]: Should we not first mention `watch`ing the `msg` as best practice (over `this.$socket.on`)?

File: docs/nodes/widgets/ui-template.md [hunk 2]
```diff
@@ -85,11 +85,23 @@ When accessing the `msg` variable inside a `<script />` tag, you need to prefix
 
 ***Important Note:*** On first load, `msg.payload` may be `null` or `undefined`, and trying to access a nested property will throw an error. Using the **optional chaining** (?.) operator, e.g. `msg.payload?.nested?.property` will prevent these errors occuring.
 
+#### Accessing Node-RED Global/Flow Context
+
+We are unable to send the full `flow`/`global` context stores from Node-RED to the Dashboard UI, as such, the best practice here is to use a "Change" node prior to the `ui-template` node to assign a `msg.<property>` to the relevant value from the `flow.` or `global.` store:
```
Comment (Steve-Mcl) [inline docs/nodes/widgets/ui-template.md:90]:
```suggestion
The `flow`/`global` context stores are not available in the Dashboard UI, as such, the best practice here is to use a "Change" node prior to the `ui-template` node to assign a `msg.<property>` to the relevant value from the `flow.` or `global.` store:
```

I am a little ineasy with "unable to". Might be better to just say is it not available?

**Review by Steve-Mcl — APPROVED** [review]
Approved with a couple of comments. Will leave for you to decide.

File: docs/nodes/widgets/ui-template.md (hunk 1 above)
Comment (joepavitt) [reply docs/nodes/widgets/ui-template.md:104]: It's a good point - there are differences though. The `watch` will fire on load, _and_ on new messages. The detailed approach here will fire only on new inputs. I will document both approaches.

**Pull request merged**